### ISSUE TITLE
Deviantart - add Story support

### ISSUE THREAD

**Issue by SomeRandomDude870** [RipMeApp/ripme#1756]
Hi, 
there is support the downloading Images from Deviantart.
I would like to be able to download Stories too.
Maybe make it so that it just downloads into a txt file or into a html, so it´s possible to add Stylesheets.

**Comment by rautamiekka** [comment]
Yes, that won't be too hard to do. The resulting HTML is completely diff between logged in and non-logged in sessions, though, which needs to be accounted for:

When logged in, you're looking at pure HTML.

When not logged in, you're looking at a little of HTML, and all the data is inside a JSON that's directly in the HTML.

**Comment by rautamiekka** [comment]
Here's an example I uploaded:

https://www.deviantart.com/rautamiekka/art/Story-upload-test-856640602

I'll link to another one from someone else when I find one that differs.

**Comment by Wiiplay123** [comment]
I had this working in my personal ripme-da-desc fork before Eclipse ruined everything, both for journal-style stories and the descriptions of images/stories.

I actually went through two iterations, first a pure .txt which didn't get any of the italics or bold, then I added html saving so it would get that kind of stuff, including links.

I have to wait for someone to actually implement Eclipse ripping (in a way that doesn't result in an IP ban from DeviantArt like the latest version of RipMe had just before the switch to Eclipse) in order for me to start work on the story/description ripping.

**Comment by Wiiplay123** [comment]
If the data is delivered as JSON from DeviantArt's end now, it would probably be better to save the original text from the JSON instead of converting to .html, for archival accuracy reasons and since encoding can get a bit weird and took me forever to get (probably) right in my version.

The stylesheets thing is a really good point though, since that's what I did with my fork before Eclipse.
Maybe have an option for each?

**Comment by metaprime** [comment]
[Mega-Thread] DeviantArt ripper is broken; yes we know -- de-duping all other DeviantArt issues to this one. https://github.com/RipMeApp/ripme/issues/2063
